ci_group/revolve2/ci_group/genotypes/cppnwin/modular_robot/v1/_body_develop.py

```diff
@@ -99,36 +87,38 @@ def __evaluate_cppn(
 
     # get rotation from output probabilities
     rotation_probs = [outputs[3], outputs[4]]
-    rotation = rotation_probs.index(min(rotation_probs))
+    rotation_index = rotation_probs.index(min(rotation_probs))
 
-    return (module_type, rotation)
+    return module_type, rotation_index
 
 
 def __add_child(
     body_net: multineat.NeuralNetwork,
     module: __Module,
-    child_index: int,
-    rotation: int,
-    grid: set[tuple[int, int, int]],
+    attachment_point_tuple: tuple[int, AttachmentPoint],
+    grid: NDArray[np.uint8],
 ) -> __Module | None:
-    forward = __rotate(module.forward, module.up, rotation)
-    position = __add(module.position, forward)
+    attachment_index, attachment_point = attachment_point_tuple
+
+    forward = __rotate(module.forward, module.up, attachment_point.rotation)
+    position = module.position + forward
     chain_length = module.chain_length + 1
 
     # if grid cell is occupied, don't make a child
     # else, set cell as occupied
-    if position in grid:
+
+    if grid[tuple(position)] > 0:
         return None
```

Comment:
here we check wether the binary matrix is > 0 to determine if a child can be mounted